File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    user_input = get_float("Change: ")
    while user_input < 0:
        user_input = get_float("Change: ")

    change(user_input)


def change(somme):
    somme_cents = int(somme * 100)
    counter = 0

    while somme_cents >= 25:
        somme_cents -= 25
        counter += 1

    while somme_cents >= 10:
        somme_cents -= 10
        counter += 1

    while somme_cents >= 5:
        somme_cents -= 5
        counter += 1

    while somme_cents >= 1:
        somme_cents -= 1
        counter += 1

    print(counter)


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int input(void);
int change(int);

int main(void)
{
    int amount = input();
    int nbDePieces = change(amount);
    printf("%i\n", nbDePieces);
}

// Récupère le montant de l'utilisateur
int input(void)
{
    int input = get_int("Change owed: ");
    while (input < 0)
    {
        input = get_int("Change owed: ");
    }
    return input;
}

// Renvoi le nombre de pièces nécessaires pour rendre la monnaie
int change(int amount)
{
    int counter = 0;

    while (amount >= 25)
    {
        counter++;
        amount -= 25;
    }

    while (amount >= 10)
    {
        counter++;
        amount -= 10;
    }

    while (amount >= 5)
    {
        counter++;
        amount -= 5;
    }

    while (amount >= 1)
    {
        counter++;
        amount -= 1;
    }
    return counter;
}
